lib.php

```diff
@@ -36,11 +36,9 @@
  * eliminates the need for an auto_prepend_file/auto_append_file.
  */
 function tool_excimer_after_config() {
-
     // TODO Temp ref: https://docs.moodle.org/dev/Login_callbacks#after_config
     // TODO Do we want to check if in upgrade/install etc.
```

Comment:
Can you please test and protect against a few more edge case states, eg during an install or upgrade. I'm fairly sure the _after_config callback doesn't fire at all under these so we get it free, in which just refine the todo

Reply:
Okay, will check these states

Reply:
It does call it when uninstalling. When I run the uninstall CLI program, it complains that it cannot save the profile (because the tables don't exist). So it is likely true for upgrading as well, but I never checked. It is not an issue for installing as auto profiling is switched off by default.

Reply:
I tested an exception throw here during upgrade, which appears it does get called during upgrade. Having the extra check here seems to prevent the issue from showing during an upgrade.
```
++ Exception calling 'tool_excimer_after_config' ++
* line 1073 of /lib/setup.php: call to tool_excimer_after_config()
* line 67 of /config.php: call to require_once()
* line 41 of /admin/cli/upgrade.php: call to require()
-->tool_excimer
++ 2021112901: Success (0.02 seconds) ++
++ Success (0.12 seconds) ++
== Setting new default values ==
Command line upgrade from 3.9.10+ (Build: 20211022) (2020061510.06) to 3.9.10+ (Build: 20211022) (2020061510.06) completed successfully.
```

Reply:
That said, did we want profiling to happen as part of these processes?

Reply:
from a business point of view I'd love to be able to see a profile of upgrades so and optimize them. So ideally yes. But also happy to just make it not explode for the moment and tackle this later.